RFunction.R

```diff
@@ -501,80 +532,70 @@ rFunction = function(data, travelcut,
-    mutate(dayRunThresh = quantile(runtime, probs = 0.95))
-  
-# add run time back to main data
-  data %<>% 
-    left_join(., eventtimes, by = c("ID", "stationary_runLts"))
+  #' Remaining Resting locations re-classified as Feeding if they are part of a
+  #' sequence of non-roosting time-points that remain stationary for an
+  #' unusually long period of time
+
+  logger.info("[5] Performing non-roosting stationary cumulative-time classification")
   
+  #' Derive required columns - check function definition for further details
+  data <- data |> add_cmltv_stationary_cols()
   
-  # Reasssign SResting => SFeeding
-  # The longest 5 percentiles of runs of stationary behaviours will be reclassified as SFeeding
+  #' Re-classify Resting locations assigned with cumulative stationary times
+  #' that exceed the 95th percentile of stationary run durations. Percentile
+  #' thresholds are individual-based and calculated from the input data
   data %<>% 
     mutate(
-      RULE = ifelse(cumtimestat > dayRunThresh & behav == "SResting", "[5] Extended stationary behaviour", RULE),
-      behav = ifelse(cumtimestat > dayRunThresh & behav == "SResting", "SFeeding", behav),
+      RULE = ifelse(!is.na(cumtimestat) & cumtimestat > dayRunThresh & behav == "SResting", "[5] Extended stationary behaviour", RULE),
+      behav = ifelse(!is.na(cumtimestat) & cumtimestat > dayRunThresh & behav == "SResting", "SFeeding", behav),
       #RULE = ifelse(cumtimestat_pctl < 0.05 & behav == "SResting", "[5] Extended stationary behaviour", RULE),
       #behav = ifelse(cumtimestat_pctl < 0.05 & behav == "SResting", "SFeeding", behav),
     ) %>%
     ungroup()
   
   # Log results
-  logger.trace(paste0("   ", sum(data$RULE == "[5] Extended stationary behaviour", na.rm = T), " locations re-classified as SFeeding"))
+  logger.trace(paste0("  |> ", sum(data$RULE == "[5] Extended stationary behaviour", na.rm = T), " locations re-classified as SFeeding"))
+  
+  
+  
   
+  ## [6] Speed-Time Classification --------------
+  
+  #' Remaining Resting locations re-classified as Feeding if the speed to next
+  #' location is greater the 97.5th percentile of the predicted stationary
+  #' speeds at that time of the day (day-hours)
   
-  #### 6. Speed-Time Reclassification -----
   logger.info("[6] Performing speed-time classification")
 
   data %<>% 
+    ungroup() %>%
     mutate(
-      RULE = ifelse(kmph > kmphPI97.5 & behav == "SResting", "[6] Exceed Speed-Time threshold", RULE),
-      behav = ifelse(kmph > kmphPI97.5 & behav == "SResting", "SFeeding", behav),
-      #RULE = ifelse(cumtimestat_pctl < 0.05 & behav == "SResting", "[5] Extended stationary behaviour", RULE),
-      #behav = ifelse(cumtimestat_pctl < 0.05 & behav == "SResting", "SFeeding", behav),
-    ) %>%
-    ungroup()
+      RULE = ifelse(!is.na(kmphPI97.5) & kmph > kmphPI97.5 & behav == "SResting", "[6] Exceed Speed-Time threshold", RULE),
```

Comment:
I think we decided to change this to kmphCI97.5  (smaller interval so more likely to get a change to feeding - bit more conservative)

Reply:
Ok, I'll implement that on the upcoming consolidation branch